### PR TITLE
Add pancreas data and output validation to functional tests

### PR DESCRIPTION
**Pull request by mckinsel**

### CODE REVIEW

File: tests/functional/test_matrix_service.py [hunk 1]
```diff
@@ -1,16 +1,15 @@
-import unittest
+import json
 import os
 import requests
-import json
 import time
+import unittest
```
Comment (calvinnhieu) [inline tests/functional/test_matrix_service.py]: For consistency:
_built-in packages_
\n
_installed packages_
\n
_local imports_

File: tests/functional/test_matrix_service.py [hunk 2]
```diff
@@ -43,32 +42,63 @@ class TestMatrixService(unittest.TestCase):
     def setUp(self):
         self.deployment_stage = os.environ['DEPLOYMENT_STAGE']
         self.api_url = f"https://{os.environ['API_HOST']}/v0"
+        self.res_dir = os.path.join(os.path.dirname(os.path.realpath(__file__)), "res")
```
Comment (calvinnhieu) [inline tests/functional/test_matrix_service.py]: thanks

File: tests/functional/test_matrix_service.py [hunk 3]
```diff
@@ -43,32 +42,63 @@ class TestMatrixService(unittest.TestCase):
...
-        request_id = self._post_matrix_service_request("loom")
+        request_id = self._post_matrix_service_request(
+            INPUT_BUNDLE_IDS[self.deployment_stage], "loom")
         # timeout seconds is increased to 600 as batch may tak time to spin up spot instances for conversion.
+        WaitFor(self._poll_get_matrix_service_request, request_id)\
+            .to_return_value(MatrixRequestStatus.COMPLETE.value, timeout_seconds=180)
+        self._analyze_loom_matrix_results(request_id, INPUT_BUNDLE_IDS[self.deployment_stage])
+
+    @unittest.skip
+    def test_matrix_service_ss2_small(self):
```
Comment (calvinnhieu) [inline tests/functional/test_matrix_service.py]: Actually this test I wrote is identical to Parth's `test_loom_output_matrix_service` (and broken). Can remove

File: tests/functional/validation.py [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+import zarr
+
+
+DSS_CLIENT = hca.dss.DSSClient()
+DSS_CLIENT.host = f"https://dss.{os.environ['DEPLOYMENT_STAGE']}.data.humancellatlas.org/v1"
+
+S3 = s3fs.S3FileSystem(anon=True)
+
+
+def calculate_ss2_metrics_direct(bundle_fqids):
```
Comment (calvinnhieu) [inline tests/functional/validation.py]: Cool! How long does this take for the default 200 bundle test `test_matrix_service_ss2`?

File: tests/functional/validation.py [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+
+        rsem_reader = csv.DictReader(io.StringIO(rsem_contents.decode()), delimiter="\t")
+
+        bundle_expression_sum = 0
+        bundle_expression_nonzero = 0
+        for row in rsem_reader:
+            bundle_expression_sum += float(row["TPM"])
+            if float(row['TPM']) != 0.0:
+                bundle_expression_nonzero += 1
+        return {"expression_sum": bundle_expression_sum,
```
Comment (calvinnhieu) [inline tests/functional/validation.py]: nit: single quotes for keys

File: tests/functional/test_matrix_service.py (hunk 3 above)
Comment (mckinsel) [reply tests/functional/test_matrix_service.py]: Okay, will remove

File: tests/functional/test_matrix_service.py (hunk 1 above)
Comment (mckinsel) [reply tests/functional/test_matrix_service.py]: So you're saying `requests` is out of order here?

File: tests/functional/validation.py (hunk 4 above)
Comment (mckinsel) [reply tests/functional/validation.py]: The whole set of tests is like 5 minutes.

File: tests/functional/validation.py (hunk 5 above)
Comment (mckinsel) [reply tests/functional/validation.py]: Are you saying you want more single quotes or fewer single quotes?

File: tests/functional/test_matrix_service.py (hunk 1 above)
Comment (calvinnhieu) [reply tests/functional/test_matrix_service.py]: Yea requests should be in the 2nd block and the other 4 in the first

**Pull request merged**